### PR TITLE
Multibyte functions

### PR DESCRIPTION
**Pull request by FilippChubukov**

### CODE REVIEW

File: src/compat/libc/include/stdlib.h [hunk 1]
```diff
@@ -193,22 +193,13 @@ static inline int mkstemp(char *path_template) {
 	(void)path_template;
 	return -1;
 }
-static inline int mbtowc(wchar_t *pwc, const char *s, size_t n) {
-	(void)pwc; (void)s; (void)n;
-	return 0;
-}
-static inline int wctomb(char *s, wchar_t wchar) {
```
Comment (anton-bondarev) [inline src/compat/libc/include/stdlib.h]: Why did you delete wctomb() from stdlib.h?

File: src/compat/libc/stdlib/Mybuild [hunk 2]
```diff
@@ -24,6 +24,7 @@ static module core {
 	source "strtoull.c"
 	source "atol.c"
 	source "itoa.c"
+
```
Comment (Deryugin) [inline src/compat/libc/stdlib/Mybuild]: Remove this line, please

File: src/compat/libc/stdlib/mblen.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(char *str, size_t max) {
+	char *tmp = str;
+	char *buffer = tmp;
+	int sum = 0;
+	int i =0;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: Our codestyle requires whitespaces after `=`, i.e. it should be `int i = 0;`

File: src/compat/libc/stdlib/mblen.c [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	while (buffer) {
+		if (i >= max) {
+			return sum;
+		}
+		sum = sum + (int)sizeof(tmp);
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: `sum +=`

File: src/compat/libc/stdlib/mbstowcs.c [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, char *mbstring, size_t max) {
+	char *tmp =  mbstring;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: Remove double whitespace, please

File: src/compat/libc/stdlib/wcstombs.c [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: Remove leading blank line

File: src/compat/libc/stdlib/wcstombs.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  wchar_t *wcstr, size_t max) {
+	wchar_t *tmp =  wcstr;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: Remove double white space, please

File: src/compat/libc/stdlib/mblen.c [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(char *str, size_t max) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: Check `str` to be not `NULL`.

File: src/compat/libc/stdlib/mbstowcs.c (hunk 5 above)
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: Do you really need this variable? You can simply use `mbstring` instead

File: src/compat/libc/stdlib/mbstowcs.c [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+	int i = 0;
+
+	while (tmp) {
+		if (i >= max) {
+			return i;
+		}
+	
+		*wcstring++ = (wchar_t)*tmp++;	
+		i++;
+
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: Remove this empty line, please

File: src/compat/libc/stdlib/wcstombs.c [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  wchar_t *wcstr, size_t max) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: Check arguments for `NULL`-pointer

File: src/compat/libc/stdlib/mbstowcs.c [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, char *mbstring, size_t max) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: Check arguments for `NULL`-pointer

File: src/compat/libc/stdlib/mblen.c [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(const char *str, size_t max) {
+	if (str == NULL) { 
+		return 0;
+	}
+	char *tmp = (char *)str;
+	char *buffer = tmp;
+	int sum = 0;
+	int i = 	0;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: Now you have a space + tab..

File: src/compat/libc/stdlib/wcstombs.c [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  const wchar_t *wcstr, size_t max) {
+	if (wcstr == NULL) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: This check should be done after declarations

File: src/compat/libc/stdlib/mbstowcs.c [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, const char *mbstring, size_t max) {
+	if (mbstring == NULL) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: This check should be done after declarations

Comment (FilippChubukov) [reply src/compat/libc/stdlib/mbstowcs.c]: actually, we don't need this check, cuz we check it in "while"

File: src/compat/libc/stdlib/wcstombs.c (hunk 13 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wcstombs.c]: actually, we don't need this check, cuz we check it in "while"

File: src/compat/libc/stdlib/mblen.c [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(const char *str, size_t max) {
+	if (str == NULL) {
```
[the remark this answers is not in the document]
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: actually, we don't need this check, cuz we check it in "while"

File: src/compat/libc/stdlib/mblen.c (hunk 12 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: missclick, fixed

File: src/compat/libc/stdlib/mblen.c (hunk 15 above)
[the remark this answers is not in the document]
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: Done

File: src/compat/libc/stdlib/wcstombs.c [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  const wchar_t *wcstr, size_t max) {
+	wchar_t *tmp = (wchar_t *)wcstr;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: You don't need this variable, just use `mbstring` inside loop.

File: src/compat/libc/stdlib/mbstowcs.c [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, const char *mbstring, size_t max) {
+	char *tmp = (char *)mbstring;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: You don't need this variable, just use `mbstring` inside loop.

File: src/compat/libc/stdlib/mbstowcs.c [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+	int i = 0;
+
+	while (tmp) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: I guess it should be `while (*tmp)`

File: src/compat/libc/stdlib/wcstombs.c [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+	int i = 0;
+
+	while (tmp) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: I guess it should be `while (*tmp)`

File: src/compat/libc/stdlib/mbstowcs.c (hunk 14 above)
Comment (Deryugin) [reply src/compat/libc/stdlib/mbstowcs.c]: Actually, this check is required. You should check `*mbstring` in loop, not `mbstring`.

File: src/compat/libc/stdlib/mblen.c [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+		if (i >= max) {
+			return sum;
+		}
+		sum = sum + (int)sizeof(tmp);
+	
+		*buffer = *tmp++;
+		i++;
+	}
+	
+	return sum;
```
[the remark this answers is not in the document]
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: I have no idea how to implement this function, may I get some advice

File: src/compat/libc/stdlib/mbtowc.c [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+int mbtowc(wchar_t *out, const char *in, size_t n) {
+	int i = 0;
+	if (in == NULL) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbtowc.c]: You also need to check that `out != NULL`, because you use this pointer.

File: src/compat/libc/stdlib/wcstombs.c [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  const wchar_t *wcstr, size_t max) {
+	int i = 0;
+	if (wcstr == NULL) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: Check `mbstr != NULL` as well.

File: src/compat/libc/stdlib/wctomb.c [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+int wctomb(char *out,  const wchar_t *in) {
+	if (in == NULL) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wctomb.c]: Checkout `out != NULL` as well.

Comment (FilippChubukov) [reply src/compat/libc/stdlib/wctomb.c]: add

File: src/compat/libc/stdlib/wcstombs.c (hunk 22 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wcstombs.c]: added

File: src/compat/libc/stdlib/mbtowc.c (hunk 21 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mbtowc.c]: added

File: src/compat/libc/stdlib/mbstowcs.c [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, const char *mbstring, size_t max) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: If `mbstring` contains terminating null character, it should be converted too, but that's not happening.

By the way, tests failed to check it, so they should be fixed as well.

File: src/compat/libc/stdlib/mbtowc.c [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+int mbtowc(wchar_t *out, const char *in, size_t n) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbtowc.c]: It looks like this function does the same stuff as `mbstowcs()`, but that's incorrect, it should write at most 1 wchar to `out`

```
mbtowc   - convert a multibyte sequence to a wide character
mbstowcs - convert a multibyte string to a wide-character string 
```

File: src/compat/libc/stdlib/wctomb.c [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+ * @date 07.12.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+int wctomb(char *out,  const wchar_t *in) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wctomb.c]: Same problem as `mbtowc`/`mbstowcs`: this function is not the same with `wcstombs`

File: src/compat/libc/stdlib/wcstombs.c [hunk 27]
```diff
@@ -0,0 +1,34 @@
...
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t wcstombs(char *mbstr,  const wchar_t *wcstr, size_t max) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: If `wcstr` contains terminating `L'\0'`, then terminating `'\0'`-char should be written to `mbstr`.

File: src/compat/libc/stdlib/wctomb.c (hunk 26 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wctomb.c]: change function

File: src/compat/libc/stdlib/mbtowc.c (hunk 25 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mbtowc.c]: changed function

File: src/compat/libc/stdlib/mblen.c [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(const char *str, size_t max) {
+	if (str == NULL) {
```
[the remark this answers is not in the document]
Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: fixed

File: src/compat/libc/stdlib/mbtowc.c [hunk 29]
```diff
@@ -0,0 +1,27 @@
...
+	if (in == NULL || out == NULL) {
+		return 0;
+	}
+	if (in == '\0') {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbtowc.c]: Always check that your code at least compiles.

```
src/compat/libc/stdlib/mbtowc.c: In function ‘mbtowc’:
src/compat/libc/stdlib/mbtowc.c:20:9: error: comparison between pointer and zero character constant [-Werror=pointer-compare]
```

File: src/compat/libc/stdlib/wcstombs.c (hunk 27 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wcstombs.c]: fix it and add test for this situation

File: src/compat/libc/stdlib/mbstowcs.c [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+
+	while (*mbstring) {
+		if (i >= max) {
+			return i;
+		}
+	
+		*wcstring++ = *mbstring++;	
+		i++;
+	}
+	if (*mbstring == '\0') {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: This condition seems to be always true.

File: src/compat/libc/stdlib/mbstowcs.c [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+size_t mbstowcs(wchar_t *wcstring, const char *mbstring, size_t max) {
+	int i = 0;
+	if (mbstring == NULL || wcstring == NULL) {
+		return 0;
+	}
+
+	while (*mbstring) {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: It's better to write `*mbstring != '\0'` implicitly.

File: src/compat/libc/stdlib/mbstowcs.c [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+		if (i >= max) {
+			return i;
+		}
+	
+		*wcstring++ = *mbstring++;	
+		i++;
+	}
+	if (*mbstring == '\0') {
+		*wcstring = *mbstring;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: I guess `i` should be increased as well.

File: src/compat/libc/stdlib/mbtowc.c [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
+
+int mbtowc(wchar_t *out, const char *in, size_t n) {
+	if (in == NULL || out == NULL) {
+		return 0;
+	}
+	if (*in == '\0') {
+		return 0;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbtowc.c]: Single null-char should be converted in this case.

File: src/compat/libc/stdlib/wcstombs.c [hunk 34]
```diff
@@ -0,0 +1,34 @@
...
+
+	while (*wcstr) {
+		if (i >= max) {
+			return i;
+		}
+	
+		*mbstr++ = *wcstr++;
+		i++;
+	}
+	if (*wcstr == '\0') {
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: It seems that this condition is always true.

File: src/compat/libc/stdlib/wcstombs.c [hunk 35]
```diff
@@ -0,0 +1,34 @@
...
+		*mbstr = *wcstr;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: `i` should be increased here.

File: src/compat/libc/stdlib/wctomb.c [hunk 36]
```diff
@@ -0,0 +1,27 @@
...
+	if (in == NULL || out == NULL) {
+		return 0;
+	}
+	if (*in == L'\0') {
+		return 0;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wctomb.c]: Single null-char should be written in this case.

File: src/compat/libc/stdlib/mblen.c [hunk 37]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+int mblen(const char *str, size_t max) {
+	int n = 5;
+	unsigned int ucs = *str & 0xFF;
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: `str` pointer is used here before check for NULL-ptr.

Comment (FilippChubukov) [reply src/compat/libc/stdlib/mblen.c]: fixed

File: src/compat/libc/stdlib/wctomb.c (hunk 36 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wctomb.c]: fixed

File: src/compat/libc/stdlib/wcstombs.c (hunk 34 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wcstombs.c]: u are right, reworked this place in wcstombs and mbstowcs

File: src/compat/libc/stdlib/wcstombs.c [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+
+	while (*wcstr != L'\0') {
+		if (i >= max) {
+			return i;
+		}
+	
+		*mbstr++ = *wcstr++;
+		i++;
+	}
+		*mbstr = *wcstr; /* Compare '\0'*/
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: Double tab. Also, why "compare"?

File: src/compat/libc/stdlib/mbstowcs.c [hunk 39]
```diff
@@ -0,0 +1,33 @@
...
+
+	while (*mbstring != '\0') {
+		if (i >= max) {
+			return i;
+		}
+	
+		*wcstring++ = *mbstring++;	
+		i++;
+	}
+		*wcstring = *mbstring; /* Compare '\0'*/
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: Double tab.

Comment (FilippChubukov) [reply src/compat/libc/stdlib/mbstowcs.c]: fixed

File: src/compat/libc/stdlib/wcstombs.c (hunk 38 above)
Comment (FilippChubukov) [reply src/compat/libc/stdlib/wcstombs.c]: fixed

File: src/compat/libc/stdlib/mblen.c [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+
+	ucs = *str & 0xFF;
+	if (ucs < 128) {
+		return 1;
+	}
+	if ((ucs & 0xC0) == 0x80 || ucs == 0xFF || ucs == 0xFE) {
+		return -1;
+	}
+
+	while (n && (ucs & (1 << n))) { /* Depending on the number of bytes,
```
Comment (anton-bondarev) [inline src/compat/libc/stdlib/mblen.c]: Move description before the loop

File: src/tests/stdlib/Mybuild [hunk 41]
```diff
@@ -12,6 +12,22 @@ module bsearch_test {
...
+}
+module wcstombs_test {
+	source "wcstombs_test.c"
+	depends embox.compat.libc.stdlib.multibyte
+	depends embox.framework.LibFramework
+}
+module mblen_test {
+	source "mblen_test.c"
+	depends embox.compat.libc.stdlib.multibyte
+	depends embox.framework.LibFramework
```
Comment (anton-bondarev) [inline src/tests/stdlib/Mybuild]: Why required this dependence ?

File: src/tests/stdlib/Mybuild [hunk 42]
```diff
@@ -12,6 +12,22 @@ module bsearch_test {
...
 
+module mbstowcs_test {
+	source "mbstowcs_test.c"
+	depends embox.compat.libc.stdlib.multibyte
+	depends embox.framework.LibFramework
+}
+module wcstombs_test {
+	source "wcstombs_test.c"
+	depends embox.compat.libc.stdlib.multibyte
+	depends embox.framework.LibFramework
```
Comment (anton-bondarev) [inline src/tests/stdlib/Mybuild]: Why required this dependence ?

File: src/tests/stdlib/Mybuild [hunk 43]
```diff
@@ -12,6 +12,22 @@ module bsearch_test {
 	depends embox.framework.LibFramework
 }
 
+module mbstowcs_test {
+	source "mbstowcs_test.c"
+	depends embox.compat.libc.stdlib.multibyte
+	depends embox.framework.LibFramework
```
Comment (anton-bondarev) [inline src/tests/stdlib/Mybuild]: Why required this dependence ?

File: src/tests/stdlib/wcstombs_test.c [hunk 44]
```diff
@@ -0,0 +1,50 @@
...
+	length = wcstombs(buffer, test_string, sizeof(check_string));
+	test_assert(length == sizeof(check_string) - 1);
+	test_assert(memcmp(buffer,check_string,length) == 0);
+}
+
+TEST_CASE("wctomb compare ABC") {
+	int length = 0;
+	char check_string[] = "A";/* Just first symbol of test_string */
+	wchar_t test_string[] = TEST_STRING;
+	char buffer[10];
```
Comment (anton-bondarev) [inline src/tests/stdlib/wcstombs_test.c]: Why 10 bytes?

File: src/tests/stdlib/wcstombs_test.c [hunk 45]
```diff
@@ -0,0 +1,50 @@
...
+	length = wctomb(buffer, test_string);
+	test_assert(length == sizeof(char));
+	test_assert(buffer[0] == check_string[0]);
+}
+
+TEST_CASE("wctombs compare terminating null character") {
+	int length = 0;
+	char check_string[] = "\0";
+	wchar_t test_string[] = L"\0";
+	char buffer[10];
```
Comment (anton-bondarev) [inline src/tests/stdlib/wcstombs_test.c]: Why 10 bytes array len?

File: src/tests/stdlib/wcstombs_test.c (hunk 44 above)
Comment (FilippChubukov) [reply src/tests/stdlib/wcstombs_test.c]: stock in case functions do something wrong

File: src/tests/stdlib/wcstombs_test.c (hunk 45 above)
Comment (FilippChubukov) [reply src/tests/stdlib/wcstombs_test.c]: stock in case functions do something wrong

File: src/tests/stdlib/mbstowcs_test.c [hunk 46]
```diff
@@ -0,0 +1,50 @@
...
+	test_assert(memcmp(buffer,check_string,length) == 0);
+}
+
+TEST_CASE("mbtowc compare ABC") {
+	int length = 0;
+	char test_string[] = TEST_STRING;
+	wchar_t check_string[] = L"A"; /* Just first symbol of test_string */
+	wchar_t buffer[10];
+
+	length = mbtowc(buffer,test_string,sizeof(char));
```
Comment (Deryugin) [inline src/tests/stdlib/mbstowcs_test.c]: Add whitespace after commas, i.e. `buffer,test` -> `buffer, test`

File: src/tests/stdlib/mbstowcs_test.c [hunk 47]
```diff
@@ -0,0 +1,50 @@
...
+
+TEST_CASE("mbstowcs compare ABC") {
+	int length = 0;
+	char  test_string[] = TEST_STRING;
+	wchar_t check_string[] = CHECK_STRING;
+	wchar_t buffer[10];
+
+	length = mbstowcs(buffer, test_string, sizeof(test_string));
+	test_assert(length == sizeof(test_string) - 1);
+	test_assert(memcmp(buffer,check_string,length) == 0);
```
Comment (Deryugin) [inline src/tests/stdlib/mbstowcs_test.c]: Add whitespace after commas

File: src/tests/stdlib/mbstowcs_test.c (hunk 46 above)
Comment (FilippChubukov) [reply src/tests/stdlib/mbstowcs_test.c]: added

File: src/tests/stdlib/mbstowcs_test.c (hunk 47 above)
Comment (FilippChubukov) [reply src/tests/stdlib/mbstowcs_test.c]: added

File: src/compat/libc/stdlib/wcstombs.c [hunk 48]
```diff
@@ -0,0 +1,35 @@
...
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: These two headers (`size_t.h`, `wchar_t.h`) are included in `stdlib.h`, so you dont need to include them.

File: src/compat/libc/stdlib/wctomb.c [hunk 49]
```diff
@@ -0,0 +1,24 @@
...
+ * @see stdlib.h
+ *
+ * @date 07.12.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wctomb.c]: These two headers (`size_t.h`, `wchar_t.h`) are included in `stdlib.h`, so you dont need to include them.

File: src/compat/libc/stdlib/mblen.c [hunk 50]
```diff
@@ -0,0 +1,41 @@
...
+ * @brief mblen function.
+ *
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mblen.c]: Unnecessary header

File: src/compat/libc/stdlib/mbstowcs.c [hunk 51]
```diff
@@ -0,0 +1,34 @@
...
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbstowcs.c]: `string.h` is not used in this file.

`size_t.h`/`wchar_t.h` are already included in `stdlib.h`

File: src/compat/libc/stdlib/mbtowc.c [hunk 52]
```diff
@@ -0,0 +1,24 @@
...
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <defines/size_t.h>
+#include <defines/wchar_t.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/mbtowc.c]: `string.h` is not used in this file.

`size_t.h`/`wchar_t.h` are already included in `stdlib.h`

File: src/compat/libc/stdlib/wcstombs.c [hunk 53]
```diff
@@ -0,0 +1,35 @@
...
+ * @brief wcstombs function.
+ *
+ * @see stdlib.h
+ *
+ * @date 01.11.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wcstombs.c]: `string.h is not used in this file.`

File: src/compat/libc/stdlib/wctomb.c [hunk 54]
```diff
@@ -0,0 +1,24 @@
...
+ * @brief wctomb function.
+ *
+ * @see stdlib.h
+ *
+ * @date 07.12.18
+ * @author Chubukov Filipp
+ */
+
+#include <stdlib.h>
+#include <string.h>
```
Comment (Deryugin) [inline src/compat/libc/stdlib/wctomb.c]: `string.h is not used in this file.`
